src/main/java/org/apereo/openequella/adminconsole/launcher/ClientLauncher.java

```diff
@@ -0,0 +1,391 @@
+      all.add(generalActions, new Rectangle(0, 0, 1, 1));
+      all.add(serverList, new Rectangle(0, 2, 1, 1));
+      all.add(serverActions, new Rectangle(0, 4, 1, 1));
+
+      updateButtons();
+
+      getContentPane().add(all);
+      getRootPane().setDefaultButton(launchButton);
+      addWindowListener(this);
+      pack();
+
+      ComponentHelper.centreOnScreen(this);
+      setVisible(true);
+
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private JComponent createServerList() {
+    final JLabel profileLabel = new JLabel(LABEL_SERVER);
+
+    final JPanel serverPanel = new JPanel(
+        new TableLayout(new int[] { profileLabel.getPreferredSize().height, serverPicker.getPreferredSize().height },
+            new int[] { TableLayout.FILL, serverPicker.getPreferredSize().width }));
+    serverPanel.add(profileLabel, new Rectangle(0, 0, 2, 1));
+    serverPanel.add(serverPicker, new Rectangle(0, 1, 2, 1));
+
+    return serverPanel;
+  }
+
+  protected JComponent createGeneralActions() {
+
+    addServerButton.addActionListener(this);
+    editProxyButton.addActionListener(this);
+
+    final int buttonWidth = ComponentHelper.maxWidth(addServerButton, editProxyButton);
+    JPanel buttonsPanel = new JPanel(new TableLayout(new int[] { addServerButton.getPreferredSize().height },
+        new int[] { TableLayout.FILL, buttonWidth, buttonWidth, TableLayout.FILL }));
+
+    buttonsPanel.add(addServerButton, new Rectangle(1, 0, 1, 1));
+    buttonsPanel.add(editProxyButton, new Rectangle(2, 0, 1, 1));
+
+    return buttonsPanel;
+  }
+
+  private JComponent createServerActions() {
+
+    editServerButton.addActionListener(this);
+    removeServerButton.addActionListener(this);
+    makeDefaultServerButton.addActionListener(this);
+    launchButton.addActionListener(this);
+
+    final int buttonWidth = ComponentHelper.maxWidth(editServerButton, removeServerButton, makeDefaultServerButton,
+        launchButton);
+
+    final JPanel serverActionsPanel = new JPanel(
+        new TableLayout(new int[] { editServerButton.getPreferredSize().height },
+            new int[] { TableLayout.FILL, buttonWidth, buttonWidth, buttonWidth, buttonWidth, TableLayout.FILL }));
+
+    serverActionsPanel.add(editServerButton, new Rectangle(1, 0, 1, 1));
+    serverActionsPanel.add(removeServerButton, new Rectangle(2, 0, 1, 1));
+    serverActionsPanel.add(makeDefaultServerButton, new Rectangle(3, 0, 1, 1));
+    serverActionsPanel.add(launchButton, new Rectangle(4, 0, 1, 1));
+
+    return serverActionsPanel;
+  }
+
+  private void updateButtons() {
+    boolean enable = serverPicker.getSelectedIndex() > -1;
+    launchButton.setEnabled(enable);
+    editServerButton.setEnabled(enable);
+    removeServerButton.setEnabled(enable);
+    makeDefaultServerButton.setEnabled(enable);
+  }
+
+  private static Config readConfig() {
+    try {
+      return Config.readServerConfigFile();
+    } catch (Exception e) {
```

Comment:
Here is a IOException, isn't it?

Reply:
It's a runtime exception, because the JSON reader throws a bunch of exceptions that I don't want to handle everywhere.